internal/pkg/aws/codestar/codestar.go

```diff
@@ -0,0 +1,49 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package codestar provides a client to make API requests to AWS CodeStar Connections.
+package codestar
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/codestarconnections"
+)
+
+type api interface {
+	GetConnection(input *codestarconnections.GetConnectionInput) (*codestarconnections.GetConnectionOutput, error)
+}
+
+// Connection represents a client to make requests to AWS CodeStarConnections.
+type Connection struct {
+	client api
+}
+
+// New creates a new CloudFormation client.
+func New(s *session.Session) *Connection {
+	return &Connection{
+		codestarconnections.New(s),
+	}
+}
+
+// WaitUntilStatusAvailable blocks until the connection status has been updated from `PENDING` to `AVAILABLE` or until the max attempt window expires.
+func (c *Connection) WaitUntilStatusAvailable(ctx context.Context, connectionARN string) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timed out waiting for connection %s status to change from PENDING to AVAILABLE", connectionARN)
+		case <-time.After(10 * time.Second):
```

Comment:
One small drawback at the moment is that we don't immediately check the status and instead first sleep 10 seconds. So that means that the tests will all at least take 10 seconds to run as well.

Instead, we can change the method body slightly so that we check immediately and then sleep for 10 seconds.
```go
var interval time.Duration // Default to 0.
for {
    select {
    ...
    case <-time.After(interval):
       ....
       interval = 10 * time.Second
    }
}
```